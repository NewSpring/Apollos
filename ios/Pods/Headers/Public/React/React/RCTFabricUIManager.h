../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h