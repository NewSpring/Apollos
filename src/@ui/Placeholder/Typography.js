import { compose, mapProps } from 'recompose';
import { StyleSheet } from 'react-native';
import Placeholder from 'rn-placeholder';
import styled from '@ui/styled';
import { Line } from './Line';

export const Typography = compose(
  mapProps(({ style, ...otherProps }) => ({
    flattenedStyles: StyleSheet.flatten(style),
    ...otherProps,
  })),
  styled(({ flattenedStyles: { fontSize, lineHeight } = {} }) => {
    const styles = { };
    if (fontSize && lineHeight) {
      styles.height = fontSize;
      styles.marginVertical = (lineHeight - fontSize) / 2;
    }
    return styles;
  }, 'Placeholder.Typography'),
)(Line);

export default Placeholder.connect(Typography);
