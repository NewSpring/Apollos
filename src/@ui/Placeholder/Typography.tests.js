import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import { Typography } from './Typography';

describe('The Typography placeholder', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Providers><Typography /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('translates type styles', () => {
    const tree = renderer.create(
      <Providers><Typography style={{ fontSize: 24, lineHeight: 32 }} /></Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
