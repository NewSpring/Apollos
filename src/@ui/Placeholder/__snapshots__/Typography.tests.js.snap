// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Typography placeholder renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#dddddd",
      "borderRadius": 6,
      "height": 18,
      "width": "100%",
    }
  }
/>
`;

exports[`The Typography placeholder translates type styles 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#dddddd",
      "borderRadius": 6,
      "height": 24,
      "marginVertical": 4,
      "width": "100%",
    }
  }
/>
`;
